apps/scan/frontend/src/app_root.tsx

```diff
@@ -228,6 +230,7 @@ export function AppRoot(): JSX.Element | null {
     return (
       <PollWorkerScreen
         electionDefinition={electionDefinition}
+        onPollsClose={() => voterSettingsControls.resetVoterSettings()}
```

Comment:
I think it would be clearer if `PollWorkerScreen` used the `useVoterSettingsControl` hook directly rather than passing in a generic `onPollsClose` handler, since we don't really need a generic handler.

That being said, the way the hook is currently written, that will also create a side effect of registering the change listeners again. I'm not sure if that would cause problems or not, but I'm a bit concerned about double side effects. I think to protect against this we should separate out the creation of the API to control the voter settings (keeping this in the hook) from the change listeners that uses the API (putting these directly in `AppRoot`).

Reply:
Great point, I agree with that. Moving the listeners out of the hook into `AppRoot` feels right. I also didn't like that the hook wasn't reusable across components without re-registering listeners, so this addresses that, as it felt like this hook was owning too much functionalit. Will push a commit soon with these changes